### PR TITLE
COMMERCE-6735 - Commerce SKU Discount Target Contribution (GSEMEASO-44)

### PR DESCRIPTION
**Pull request by janbrychtadotcom**
- headless-commerce-admin-pricing: add methods DiscountSku
- commerce-admin-pricing - add methods: getCPInstancesByCommerceDiscountId, getCPInstancesByCommerceDiscountIdCount

### CODE REVIEW

**Comment by marco-leo** [comment]
ci:test:sf

**Comment by marco-leo** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/dto/v2_0/converter/DiscountSkuDTOConverter.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			commerceDiscountRel.getCommerceDiscount();
+
+		return new DiscountSku() {
+			{
+				actions = dtoConverterContext.getActions();
+				discountExternalReferenceCode =
+					commerceDiscount.getExternalReferenceCode();
+				discountId = commerceDiscount.getCommerceDiscountId();
+				discountSkuId = commerceDiscountRel.getCommerceDiscountRelId();
+				skuId = cpInstance.getCPInstanceId();
```
Comment (riccardo-alberti) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/dto/v2_0/converter/DiscountSkuDTOConverter.java:66]: @janbrychtadotcom let's return also the skuExternalReferenceCode. So it is consistent with the other targets.

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-test/src/testIntegration/java/com/liferay/headless/commerce/admin/pricing/resource/v2_0/test/DiscountSkuResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.junit.Ignore;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Zoltán Takács
+ */
+@Ignore
+@RunWith(Arquillian.class)
+public class DiscountSkuResourceTest extends BaseDiscountSkuResourceTestCase {
+}
```
Comment (riccardo-alberti) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-test/src/testIntegration/java/com/liferay/headless/commerce/admin/pricing/resource/v2_0/test/DiscountSkuResourceTest.java:28]: @janbrychtadotcom  we should implement the test cases to cover the endpoints exposed here. thnx

**Review by riccardo-alberti — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-discount-api/src/main/java/com/liferay/commerce/discount/constants/CommerceDiscountConstants.java [hunk 3]
```diff
@@ -74,6 +74,8 @@
 
 	public static final String TARGET_PRODUCTS = "products";
 
+	public static final String TARGET_SKU = "products-skus";
```
[the remark this answers is not in the document]
Comment (dsitu) [reply modules/apps/commerce/commerce-discount-api/src/main/java/com/liferay/commerce/discount/constants/CommerceDiscountConstants.java:77]: yep, those changes sound good to me.

File: modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/frontend/CommerceDiscountCPInstanceClayTableDataSetDisplayView.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+//		imageField.setContentRenderer("image");
+
+//		ClayTableSchemaField nameField =
+//			clayTableSchemaBuilder.addClayTableSchemaField(
+//				"product.name.LANG", "name");
+//
+//		nameField.setContentRenderer("actionLink");
+//
+//		clayTableSchemaBuilder.addClayTableSchemaField("product.sku", "sku");
+
```
Comment (dsitu) [inline modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/frontend/CommerceDiscountCPInstanceClayTableDataSetDisplayView.java:62]: We should avoid comments when possible.

File: modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/display/context/CommerceDiscountDisplayContext.java [hunk 5]
```diff
@@ -317,6 +322,30 @@ public String getDiscountCPDefinitionApiURL() throws PortalException {
 			false);
 	}
 
+	public List<ClayDataSetActionDropdownItem>
+		getDiscountCPInstanceClayDataSetActionDropdownItems()
+		throws PortalException {
+//TODO: JB
```
Comment (dsitu) [inline modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/display/context/CommerceDiscountDisplayContext.java:328]: tab
```suggestion
		//TODO: JB

**Comment by dsitu** [comment]
@janbrychtadotcom for the first commit we should separate out the generated code into its own commit, so we can see the manual changes clearer and if we ever need to regenerate over the regen commit we can.

**Pull request closed without merging**